osc.c

```diff
@@ -449,7 +449,8 @@ static const char * device_name_check(const char *name)
 	if (!dev)
 		return NULL;
 
-	return iio_device_get_name(dev) ?: iio_device_get_id(dev);
+	return iio_device_get_label(dev) ?:
+		(iio_device_get_name(dev) ?: iio_device_get_id(dev));
```

Comment:
There's an helper already... `get_iio_device_label_or_name()`.  Using the id should be very unlikely (I think I never saw a driver not using iio_dev->name). Anyways either extend the helper or use it to make the ternary simpler :)

Reply:
I didn't notice the helper, which is much better, yes. ID is unlikely to be used indeed and if it did, it would break a lot of stuff anyway. One example that comes in mind is when osc saves attributes for a device in .ini file. It wouldn't be reliable to save for iio:device0 which might point to a different hardware after a reboot.
Thanks for the suggestion!